### DISCUSSION TITLE
Page Sizing

### DISCUSSION THREAD

**Discussion by jgoll7** [jgoll7/Tafe22S211#8] [General]
Just wondering if you have ensured that your feature is able to fit different screen sizes?

**Comment by MoonlitBeans** [comment]
No, I didn't even think of doing that. I resized my window and it broke. I'll read up on how to fix that in the XAML code.

**Reply by jgoll7** [reply]
I used a grid with row and column definitions for mine and that seems to have worked out quite well, is that how you did yours? I can provide an example if you need

**Reply by MoonlitBeans** [reply]
I just dragged and dropped. Could you please provide an example of the XAML code for that?

**Comment by jgoll7** [comment]
Sure! 
		<Grid.ColumnDefinitions>
			<ColumnDefinition Width="*" />
			<ColumnDefinition Width="*" />
			<ColumnDefinition Width="*" />
			<ColumnDefinition Width="*" />
			<ColumnDefinition Width="*" />
		</Grid.ColumnDefinitions>
		<Grid.RowDefinitions>
			<RowDefinition Height="120" />
			<RowDefinition Height="*" />
			<RowDefinition Height="*" />
			<RowDefinition Height="*" />
			<RowDefinition Height="*" />

		</Grid.RowDefinitions>

this will create 5 rows and 5 columns. You can either set the width and height to auto(takes up as much space as the elements inside need), a specific value (120 in this example) or * which will give all remaining space to that column/definition. 

You then define which row and column you want your controls to go in, for example:
<Button HorizontalAlignment="Center" Grid.ColumnSpan="3" Grid.Column="1" Grid.Row="1" Grid.RowSpan="3" Width="500">
This will place the button on column 1 (2nd column as 0 is the 1st) and row 1, use row span and column span to determine if the control needs to span across multiple columns/rows.

Hope this helps in some way, I think my xaml is a bit messy so I might need to go over this stuff.

**Reply by jgoll7** [reply]
for some reason when looking at the code it doesnt show it for me, it looks like this: 
![definitions](https://user-images.githubusercontent.com/112138014/189129221-90715809-1e1a-44ee-921d-0046dc5c666d.png)
